px4_ros2_cpp/src/components/mode_executor.cpp

```diff
@@ -290,6 +290,36 @@ void ModeExecutorBase::arm(const CompletedCallback & on_completed)
     [this](const px4_msgs::msg::VehicleStatus::UniquePtr & msg) {return _is_armed;}, on_completed);
 }
 
+void ModeExecutorBase::disarm(const CompletedCallback & on_completed, const bool & forced)
+{
+  if (!_is_armed) {
+    on_completed(Result::Success);
+    return;
+  }
+
+  Result result;
+
+  if (forced)
```

Comment:
This would be a bit simpler:
```
const float param2 = forced ? 21196.f : NAN;
const result = sendCommandSync(px4_msgs::msg::VehicleCommand::VEHICLE_CMD_COMPONENT_ARM_DISARM,
    0.f, param2);

```